util.h

```diff
@@ -83,4 +85,25 @@ checked_calloc(size_t nmemb, size_t size, const char *filename, unsigned int lin
 #define malloc2(size)        checked_malloc((size), __FILE__, __LINE__, __func__)
 #define calloc2(nmemb, size) checked_calloc((nmemb), (size), __FILE__, __LINE__, __func__)
 
+/* Data-loading definitions */
+static inline const char *
+get_data_file(const char *filename)
+{
+  struct stat s;
+
+  if (stat(filename, &s) == 0) {
+    return filename;
+  }
+
+#ifdef DATA_DIR
+  char *data_dir_filename = malloc(strlen(DATA_DIR) + 1 + strlen(filename) + 1);
+  sprintf(data_dir_filename, "%s/%s", DATA_DIR, filename);
+  if (stat(data_dir_filename, &s) == 0) {
+    return (const char *)data_dir_filename;
+  }
+#endif
+
+  return NULL;
+}
+
```

Comment:
This way, we are leaking the malloc()'d pointers. It's no big deal since it's a one-time call in the _init functions, but it still makes me a bit nervous. I'd just have a static buffer and return a pointer to that. But it's nitpicking, feel free to ignore.